src/opm/input/eclipse/Schedule/Network/ExtNetwork.cpp

```diff
@@ -86,24 +86,65 @@ void ExtNetwork::add_branch(Branch branch)
     this->m_branches.push_back( std::move(branch) );
 }
 
-void ExtNetwork::drop_branch(const std::string& uptree_node, const std::string& downtree_node) {
-    auto downtree_branches = this->downtree_branches( downtree_node );
-    if (downtree_branches.empty()) {
+void ExtNetwork::add_or_replace_branch(Branch branch)
+{
+    const std::string& uptree_node = branch.uptree_node();
+    const std::string& downtree_node = branch.downtree_node();
+
+    // Add any missing nodes
+    for (const std::string& nodename : { downtree_node, uptree_node }) {
+        if (!this->has_node(nodename)) {
+            this->m_nodes.insert_or_assign(nodename, Node{nodename});
+        }
+        if (!this->has_indexed_node_name(nodename)) {
+            this->add_indexed_node_name(nodename);
+        }
+    }
+    
+    // Remove any other branch uptree from downtree_node (gathering tree structure required)
+    auto uptree_link = this->uptree_branch( downtree_node );
```

Comment:
Ah, you're right - might as well remove the whole 'update' block actually?

Reply:
It is design/taste dependent. 

> might as well remove the whole 'update' block actually?

If we do this, maybe we can add some comment before drop_branch to explain if the `[uptree_node, downtree_node]` is the same with `(old_uptree_node, downtree_node);`, we will add it back with the updated properties later.  Or something similar.  It is just for later ease to understand, because there are multiple different ways to achieve the same thing here.

Reply:
Agreed, will add a comment for clarity while reducing the amount of code.